### PR TITLE
[MRG] Fix not positive definitive matrix in parallel optimisation

### PR DESCRIPTION
**Pull request by betatim**
This does two things: tries to fix the Circlie CI error we keep seeing about "not pos def matrix" in the parallel evaluation example and makes the calls with a parallel strategy reproducible by fixing the seed.

### CODE REVIEW

**Comment by iaroslav-ai** [comment]
Fine with me as a solution to current circleci failures

**Comment by iaroslav-ai** [comment]
Ahaaaaaaaaa so the difference to sklearn implementation is in that there is a stabilization of matrix inverse using alpha > 0.0 
Consider adding this as a test (fails when alpha is set to 0.0):
```python
def test_gpr_handles_similar_points():
    """
    This tests whether our implementation of GPR 
    does not crash when the covariance matrix whose
    inverse is calculated during fitting of the 
    regressor is singular. 
    Singular covariance matrix often indicates
    that same or very close points are explored 
    during the optimization procedure.
    """

    import numpy as np
    from skopt.learning.gaussian_process.gpr import GaussianProcessRegressor
    
    X = np.random.rand(8, 3)
    y = np.random.rand(8)
    
    X[:3,:] = 0.0
    y[:3] = 1.0

    model = GaussianProcessRegressor()
    # this fails if singular matrix is not handled
    model.fit(X, y)
```

**Comment by iaroslav-ai** [comment]
So LGTM

**Comment by betatim** [comment]
Test added.

**Comment by betatim** [comment]
Merging this.

**Comment by iaroslav-ai** [comment]
Nice. That was a long standing issue
